DockDoor/Utilities/KeybindHelper.swift

```diff
@@ -110,18 +110,14 @@ class KeybindHelper {
 
     private func showHoverWindow() {
         Task { [weak self] in
-            do {
+            guard let self else { return }
+            await MainActor.run { [weak self] in
                 guard let self else { return }
-                let windows = try await WindowUtil.activeWindows(for: "")
```

Comment:
The activeWindows call to the empty string also serves to update the space cache, I don't think we need to change this part.

Reply:
What cache update do you mean?

Reply:
For tracking windows in all spaces, the call to activeWindows is needed here.

Reply:
Didn't calling `activeWindows` with an empty string simply return all that was in the cache?
https://github.com/ejbills/DockDoor/blob/31f8c1b2c8f478e9f85d1dea692f98db715e56b9/DockDoor/Utilities/WindowUtil.swift#L409-L411

Reply:
It did indeed return `getAllWindows`, but not till after it had made a few calls to `fetchWindowInfo` which were called before the return (even with an empty string). And the `fetchWindowInfo` func served to also update the cache [here.](https://github.com/ejbills/DockDoor/blob/31f8c1b2c8f478e9f85d1dea692f98db715e56b9/DockDoor/Utilities/WindowUtil.swift#L473)

Reply:
Okay, I got it. I don't think it should be done implicitly, I'll add a separate "cache refresh" function and call it explicitly.

Reply:
Fair enough

Reply:
I added in the explicit call to updating the methods as discussed, so this discussion is resolved.